### ISSUE TITLE
Self-Supervised Deep Learning to Enhance Breast Cancer Detection on Screening Mammography

### ISSUE THREAD

**Issue by chrstnkgn** [iRAIL-SNU/irail-paper-review#17]
### Motivation

- 의료 분야에서 대규모의 정제된 데이터셋을 만드는 것은 시간적/경제적 비용이 많이 드는 일이다. (label scarcity)
    - 이를 해결하기 위해 나온 방법론들 중 하나가 self-supervised learning (SSL) → combination of semi supervised and transfer learning
- Implications of the study
    - mammogram에 suitable한 transformation에 대해 탐색했다.
    - patch로 쪼개 pretrain을 진행하는 방식은 효과적인 feature learning을 제공함과 동시에 연산량을 줄여주는 효과를 준다.
    - Pooling layer에서 간단한 attention기법을 사용하면 classification performance를 상승시킬 수 있다 (다만 fine tuning에서는 그 상승폭이 좁다.)
    - 4개의 SSL 기법 중 SWaV의 성능이 가장 좋았다.
    - 서로 다른 데이터셋 간에서도 이 pretrained model은 trasferable함을 보여줬다

### Method

0. SSL 방법론
    - **SimCLR:** Strong augmentation을 활용해서 같은 이미지의 서로 다른 augmentation을 positive pair로 정의하고 다른 이미지에서 나온 augmentation 결과를 negative pair로 정의해서 contrastive learning을 진행하는 방식.
    - **BYOL:** 두 개의 다른 네트워크 (online/target)를 이용해 contrastive learning 진행.
    - **SWaV**: 같은 이미지에서 나온 서로다른 augmentated image들에 cluster를 할당, 비슷한 이미지들을 군집화시켜서 군집끼리 비교.
    - **ViT-MAE**: 이미지를 패치로 쪼개어 일정부분을 마스킹해 ViT에 적용.
    - mammogram에 적절한 augmentation strategy를 찾고, large batch size의 부담을 덜어야 함 → 이미지를 패치로 쪼개서 pretraining을 진행한 후 whole image model로 컨버트하여 학습
    - GAP (Gradient Average pooling)과 MIP (Multiple instance Pooling) 간의 성능 차이 비교 (MIP는 attention 적용)

      ![image](https://user-images.githubusercontent.com/101173633/222657330-03a18d55-ad1e-4987-bf2f-5944868797fe.png)

1. Dataset 
    - CBIS-DDSM
        - 환자: 1566, 이미지: 3119, cancer case: 761 (환자기준)
        - 모든 positive case는 ROI annotated
        - Split: 7:1:2
        - size: 1152*896
        - Annotated patch 생성: 하나는 ROI를 center로 해서, 또하나는 랜덤으로  (224*224)
            - categories: background, malignant mass, benign mass, malignant calcification, benign calcification
    - CMMD
        - 환자: 1775, 이미지: 3728
        - 모든 환자가 tumor 가지고 있음 → malignant or benign
        - no ROI
        - 3:1:1 split
        - 1147*957
    - Tiled patch set
        - grid pattern over the whole image, 50% overlap, no label
        - background 20% 이상인 이미지는 제외: boundary information을 줄이기 위해서 (텍스쳐 패턴에 대해 더 집중하기 위해)
        - 64x64, 96x96, 128x128, 256x256 (제일 성능이 잘 나오는 사이즈로 선정)

### Experiment & Result

2. Annotated patch classification
    - mammogram에 제일 잘 workimg하는 transformation 탐색
        - 통상적으로 nature image에서 이용하는 strategy들은 의료영상에 적용되지 못하는 경우가 많음
        - 8개의 transformation 고려: 1개 적용 / 2개 적용 후 비교
        
        ![image](https://user-images.githubusercontent.com/101173633/222657746-977e2022-71a7-4ec9-a74e-18a1704a43d7.png)
        
        - BYOL 사용: 쉬운 implementation, batch size에 insensitive함, 이 태스크에서 성능이 잘 나오는 편에 속함
        - 결과
        
        ![image](https://user-images.githubusercontent.com/101173633/222657773-56b2f283-991b-498c-8d70-ac5f99321467.png)
        
        - Final choice: crop, gamma, contrast
    - SSL model comparison (linear evaluation)
        - ViT-MAE에서는 transform 적용하지 않음
        
        ![image](https://user-images.githubusercontent.com/101173633/222657829-fb7e6b67-8cfa-4122-99ee-e79c8d0dc6e9.png)
        
        - ***“ViT based models may be poorly suited for mammographic image classification tasks”***
        
3. CBIS-DDSM tiled patch pretraining to whole image classification
    - pretraining on tile patches → linear evaluation on whole image (패치는 라벨 없음. 이미지는 positive or negative 라벨)
    - Patch size selection
        - BYOL 사용, batch 4096 (256*256만 2048)
        
        ![image](https://user-images.githubusercontent.com/101173633/222657892-5f9007e4-fba3-4262-9647-5d4f6a6880db.png)
        
    - SSL model comparison
        - Transformer 기반 모델인 ViT-MAE는 따로 split을 진행하지 않고 사용
        
        ![image](https://user-images.githubusercontent.com/101173633/222657923-26bb1e44-402e-4199-9630-7530c1bef3e1.png)
        
        - Linear evaluation using GAP
            - SWaV가 제일 좋은 성능을 보임 (저자들의 의견: annotation patch보다 tiled patch의 imbalance가 높아서, clustering based approach가 instance based approach보다 더 잘 working했던 것 같다)
            - ViT-MAE가 제일 낮은 성능을 보임
        - GAP layer 제거하고 attention을 사용한 MIP layer 사용해서 실험 (BYOL과 SWaV 비교)
            - 전체적인 성능 향상, SWaV가 더 높은 폭으로 상승함을 보임
                
                ![image](https://user-images.githubusercontent.com/101173633/222657963-3b28ada1-55ec-447c-845a-c2d5dabdc585.png)
                
    - Finetuning
        - BYOL / SWaV 비교
        - Baseline으로 scratch에서 학습된 ResNet50 이용
        - finetuning 시에는 random flip/rotation/brightness adjustment 사용 (SSL과 다른 augmentation)
        - Finetuning 성능이 linear evaluation에 비해 상승했음을 볼 수 있음
        - 상승 폭은 GAP가 MIP에 비해 더 큼
        
        ![image](https://user-images.githubusercontent.com/101173633/222658000-447756a4-d1ff-48cb-a367-ac9e6cf21e81.png)
        
    - Alternative designs
        - 모델 크기 비교 (Resnet50 vs ResNet152)
        - 성능에 큰 차이 없다고 확인함
        
        ![image](https://user-images.githubusercontent.com/101173633/222658035-48289dc2-90db-4228-b788-2f388f0c9364.png)
        
4. SWaV Prototypa analysis
    - Image cluster pattern을 분석하고자 시행
    
    ![image](https://user-images.githubusercontent.com/101173633/222658066-75aa3fb5-706c-410e-b014-93f08eeba0e4.png)
    
    - malignant calcification (13, 18, 19)과 malignant mass (5, 6, 14, 18) 가 집중되는 prototype 들이 존재했음
    - benign과 malignant 클래스들은 비슷한 histogram을 가지고 있었음
5. CMMD training and transfer learning
    - SWaV 모델을 CMMD 데이터셋에 적용해서 실험
    
    ![image](https://user-images.githubusercontent.com/101173633/222658100-2f08df83-edb3-41cc-88ed-ee2d9001af71.png)
    
    - CMMD 데이터셋은 DDSM과 확실히 다른 intensity profile을 가지고 있음
    - CMMD로 pretrain한 경우, DDSM으로 pretrain 한 경우, GAP/MIP 사용한 경우로 비교
    
    ![image](https://user-images.githubusercontent.com/101173633/222658138-ffdebada-ab33-4f68-83e3-599fcd37507d.png)
    
    - DDSM/CMMD 성능에 큰 차이를 보이지 않음
    - SSL 모델이 데이터셋에 굉장히 transferable하다는 것을 보여줌

### Discussion
- BYOL은 annotated patch classification에서 제일 좋은 성능을 보였는데, 이는 data imbalance 차이에서 온 것으로 추정된다. 대체로 imbalance가 강한 데이터셋에서 SWaV가 강세를 보인다. 이는 cluser based 모델은 background에서 abnormality를 찾는 데에 더 특화되어 있음을 시사한다고 생각할 수 있다. 이는 cluster의 distribution을 visualize해서 보았을 때도 확인할 수 있었다.
- SimCLR 기법은 tiled patches를 이용해서 feature를 학습하는데 효과적인 방법이 아닐 수 있다. positive/negative pair 간에 큰 차이가 없다는 mammogram의 특징 때문일 수 있다.
- ViT-MAE기법이 제일 낮은 성능을 보였다. self attention이 국조적인 eature을 잡아내는데 취약함을 시사한다고 생각할 수 있다. 또한 lesion은 주변과의 spatial relationship이 별로 없기 때문일 수 있다. mammography 관련 task에서는 CNN 구조가 제일 적합하다고 생각할 수도 있을 것이다.
- 여타 global clustering method을 사용해서 실험해볼 수 있을 것이다.

### Insight

- CNN 구조를 이용한 SSL이 mammogram에 잘 working한다는 결론을 낸 논문.
- 다양한 방법론을 비교해본 것이 인상적이었음.
- 공개 데이터셋을 이용한 pretrain 결과를 서울대병원 데이터셋에 적용해서도 높은 성능을 낼 수 있을 것인지 궁금해졌음.

### Paper URL

https://arxiv.org/abs/2203.08812
